source/mqtt_agent.c

```diff
@@ -718,6 +727,42 @@ static MQTTStatus_t createAndAddCommand( CommandType_t commandType,
 
 /*-----------------------------------------------------------*/
 
+static void concludeCommand( MQTTAgentContext_t * pAgentContext,
+                             Command_t * pCommand,
+                             MQTTStatus_t returnCode,
+                             uint8_t * pSubackCodes )
+{
+    bool commandReleased = false;
+    MQTTAgentReturnInfo_t returnInfo;
+
+    ( void ) memset( &returnInfo, 0x00, sizeof( MQTTAgentReturnInfo_t ) );
+    assert( pAgentContext != NULL );
+
+    returnInfo.returnCode = returnCode;
+    returnInfo.pSubackCodes = pSubackCodes;
+
+    if( pCommand != NULL )
```

Comment:
Why is sending a NULL command acceptable to the function if the fuction's is primarily for release a command object?

Reply:
This shouldn't ever be NULL. However, since the ack list is part of the agent context struct, it's visible to the user. I figured I might as well add a check here in case the user for whatever reason decided they wanted to corrupt the array.

Reply:
If the user happens to corrupt the array, that wouldn't be considered as the correct usage of the library, would it?  Wouldn't having an assert check somewhat deter the user from meddling with the Agent context?